### DISCUSSION TITLE
Unsure on how exactly to add a script tag

### DISCUSSION THREAD

**Discussion by ZenaMel** [facebook/docusaurus#9991] [Q&A] [answered]
On the themeConfig->metadata object, I'm unsure on how to add the following script tag:

```js
    <script id="usercentrics-cmp" src="https://app.usercentrics.eu/browser-ui/latest/loader.js" data-settings-id="ID" async></script>
```

My attempt (not sure if this is correct at all):
```js
 {
    tagName: 'script',
    async: true,
    attributes: {
      id: 'usercentrics-cmp',
      src: 'https://app.usercentrics.eu/browser-ui/latest/loader.js',
      'data-settings-id': 'ID',
    }
  },
```

What would be the correct way for adding a script tag that also has an id and data attribute? Would this be correct?

**Comment by slorber** [comment]
We have multiple ways to do that, but what makes you think this one is incorrect?

Isn't the tag in the output HTML when you add it this way?

**Reply by ZenaMel** [reply]
> Isn't the tag in the output HTML when you add it this way?

@slorber No.

**Reply by slorber** [reply]
Prove it, don't put the burden on me

**Reply by ZenaMel** [reply]
@slorber I checked the HTML output and it's simply not contained in the <head> tag. I don't know what else to do to "prove" it. The method works in general, but for the script I'm trying to add it does not add it. Metatags are added no problemo.

**Reply by ZenaMel** [reply]
I also tried this to no avail:

```js
  {
    tagName: 'script',
    innerHTML: `
    id="usercentrics-cmp" src="https://app.usercentrics.eu/browser-ui/latest/loader.js" data-settings-id="ID" async
    `,
    attributes: {
      async: true,
      id: 'usercentrics-cmp',
      src: 'https://app.usercentrics.eu/browser-ui/latest/loader.js',
      'data-settings-id': 'ID',
    }
  },
```

**Reply by ZenaMel** [reply]
I think I know what the issue is now.
From what I understand script tags need to be placed under the `headTags` property, not the `metadata`. Is that correct?
https://docusaurus.io/docs/next/seo#global-metadata

**Reply by slorber** [reply]
That's the point of proving it: you figure out things on your own by being more careful while creating a https://docusaurus.new repro.

I knew what the problem was, but what matters to me is that you adopt the right mindset for reporting issues.

You have to double-check the doc, and provide a runnable code example. Otherwise I won't be helpful, because you put the burden on me to interpret your text and create the that repro for you. My time is better spent implementing features and fixing actual bugs rather than helping a single user.

**Reply by ZenaMel** [reply]
@slorber I tested it more thoroughly and I've found out the following:
(_Docusaurus v3.1.1_)

The method shown [here](https://docusaurus.io/docs/next/seo#global-metadata) does not automatically add the defined `headTags` to the `<head>` on custom pages. There's no notice on this from what I can tell. Could we perhaps add [this info](https://stackoverflow.com/a/58349826) (if not deprecated)? (it does work after testing it myself)

The solution was to create a custom component that makes use of `docusaurus/Head` and wrapping the `Layout` with it.
This feels a bit too cumbersome and is probably not the intended solution.

I'll try to get a repro running when I find the time for it.

Component.js
```js
import React from 'react';
import Head from '@docusaurus/Head';

export default function GlobalHead() {
    return (
        <Head>
            {/* Cookie Consent */}
        <script id="usercentrics-cmp" src="https://app.usercentrics.eu/browser-ui/latest/loader.js" data-settings-id="..." async></script>
        {/* Web Analytics */}
        <script defer data-domain="..." src="https://plausible.io/js/script.js"></script>
        </Head>
    )
}
    ...
```

Swizzled Layout component (`src\theme\Layout\index.js`)
```js
import React from 'react';
import Layout from '@theme-original/Layout';
import GlobalHead from '@site/src/components/GlobalHead';

export default function LayoutWrapper(props) {
  return (
    <>
      <GlobalHead />
      <Layout {...props} />
    </>
  );
}
```

Follow-up question: how global is the `global-metadata`? --> does it get inherited in the docs plugin by default? Based on the docs, I'd say yes.

**Reply by slorber** [reply, marked as answer]
If you think there's a bug, show it with a repro. 
It's possible there is a bug, but unless a repro is provided it's not proven.

Afaik headTags are not a `themeConfig` attribute but a `siteConfig` attribute. It might just be a docs typo on the SEO docs page.
https://docusaurus.io/docs/next/api/docusaurus-config#headTags

`themeConfig.metadata` should work

`<Head>` is not deprecated. You can achieve the same result in multiple ways according to your need. `<Head>` is the most granular and can be applied on a per-page basis

**Reply by ZenaMel** [reply]
@slorber Will do, thanks! I'll put in a PR in case the docs do in fact got a typo.

**Reply by slorber** [reply]
Great

Apparently, we have an issue for that doc problem already:
https://github.com/facebook/docusaurus/issues/9792

This doc page is not new, not sure why it only surfaces now
